### PR TITLE
Fixes strict-component-boundaries

### LINKED ISSUE #156: strict-component-boundaries possibly flagging incorrectly
I can't tell if the `strict-component-boundaries` rule is working as intended in this case. These are utils, not components.

Here's the [errors in CI](https://circleci.com/gh/Shopify/images/29?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) for this [code](https://github.com/Shopify/images/pull/10/commits/d36cf94db2fafb60fc743ae1dbf9eaf884db039e). 

If this isn't a bug, please close this.
cc/ @cartogram

### PR DESCRIPTION
**Pull request by cartogram**
This PR resolves false positives with strict-component-boundaries that were introduced with https://github.com/Shopify/eslint-plugin-shopify/pull/140. 

Fixes https://github.com/Shopify/eslint-plugin-shopify/issues/156 https://github.com/Shopify/eslint-plugin-shopify/issues/122 and https://github.com/Shopify/eslint-plugin-shopify/issues/123

### Problem:
The check for whether there is a component in the path was returning true if the path to a module included a pascal case word. In https://github.com/Shopify/eslint-plugin-shopify/pull/140 this check changed from the ImportDeclaration source to the full path to the module on the users computer. Since `Shopify` is a pascal case word and in most cases there is at least one pascal case word in the full path as in `Users/your-username/path`, this was returning true in almost every ImportDeclaration.

### Fix:
The solution was to check the import source for the component in path check rather than the full path to the module source.

### 🎩 Instructions:

1. Pull this branch
1. Install yalc: `yarn add yalc -g`
1. run `yalc publish`
1. cd into a project that is currently reporting false positives for this rule (such as shopify/images)
1. install this branch using yalc: `yalc add eslint-plugin-shopify`
1. confirm that the false positives are no longer invalid

### CODE REVIEW

**Comment by BPScott** [comment]
I'm not going to pretend to know what this is doing under the hood but adding a bunch of examples to a test suite would make me feel better about it working

**Comment by cartogram** [comment]
@BPScott thanks for pointing that out. I added what would be a test for this fix. However, this new test is pulling from a fixture folder and we are already ignoring when `fixtures` is in the path so the test will already never fail. 

I am going to need to go back and rework the `fixtures` ignore and update this PR. Changing the title to WIP in the mean time.

**Comment by cartogram** [comment]
Updates:
1. solved fixture import so the added test is legit (it will fail without these changes)
1. successful 🎩 in `shopify/images` and `polaris-react`

File: tests/lib/rules/strict-component-boundaries.js [hunk 1]
```diff
@@ -57,5 +62,10 @@ ruleTester.run('strict-component-boundaries', rule, {
       parserOptions,
       errors,
     },
+    {
+      code: `import someThing from '../SomeOtherComponent/fixtures/SomeMockQuery/query.json';`,
+      parserOptions,
+      errors,
+    },
```
Comment (lemonmade) [inline tests/lib/rules/strict-component-boundaries.js]: I feel like you are not testing the main thing you indicated this PR was to address

File: lib/rules/strict-component-boundaries.js [hunk 2]
```diff
@@ -17,19 +17,22 @@ module.exports = {
       ImportDeclaration(node) {
         const importSource = node.source.value;
         const resolvedSource = resolve(importSource, context);
-        const pathParts = resolvedSource
-          ? pathSegmantsFromSource(resolvedSource)
-          : pathSegmantsFromSource(importSource);
+        const pathParts = pathSegmantsFromSource(importSource);
```
Comment (lemonmade) [inline lib/rules/strict-component-boundaries.js]: In my opinion, this is not the most useful way to fix this issue. I would rather you never reference the image source, and instead operate on the relative source from the root of the app. Otherwise, I don’t think you’ll ever be able to get this to actually reflect what we want to enforce

**Review by lemonmade — CHANGES_REQUESTED** [review]

File: tests/lib/rules/strict-component-boundaries.js [hunk 3]
```diff
@@ -11,51 +11,62 @@ const parserOptions = {
 const errors = [
   {
     type: 'ImportDeclaration',
-    message: 'Strict component boundaries.',
+    message: `Do not reach into an individual component's folder for nested modules. Import from the closest shared components folder instead.`,
```
Comment (cartogram) [inline tests/lib/rules/strict-component-boundaries.js:14]: cc @GoodForOneFare with regards to https://github.com/Shopify/eslint-plugin-shopify/issues/122

**Comment by cartogram** [comment]
Update: 

- This rule now operates only on the full relative source. There are fewer edge cases when doing it this way. 
- This meant adding a "filename" property to each test that resolves to an actual fixture file. Without this, the `resolvedSource` would return undefined/missing.
- I also updated the error message to be less vague. https://github.com/Shopify/eslint-plugin-shopify/issues/122

**Review by BPScott — COMMENTED** [review]
Nice, working on absolute paths feels like it protects against false-positives when people do odd things with convoluted relative paths (e.g [this](https://github.com/Shopify/eslint-plugin-shopify/issues/119#issuecomment-409695224) should now consistently pass both examples).

Can we add an extra three test cases:

- when a component imports a sibling component using a not the simplest relative path
- when a component's subcomponents import a sibling subcomponent
- when a subcomponent imports its parent's SCSS

The following tests should all pass:

```js
{
  code: `import {someThing} from '../../components/Bar';`,
  parserOptions,
  filename: fixtureFile('basic-app/app/components/Foo/index.js'),
},
{
  code: `import {someThing} from '../Baz';`,
  parserOptions,
  filename: fixtureFile('basic-app/app/components/Foo/components/Bar/index.js'),
},
{
  code: `import {someThing} from '../../Foo.scss';`,
  parserOptions,
  filename: fixtureFile('basic-app/app/components/Foo/components/Bar/index.js'),
}
```

**Comment by cartogram** [comment]
@BPScott updated with the tests you posted.

**Review by BPScott — APPROVED** [review]
